server/webhook.go

```diff
@@ -169,33 +169,87 @@ func newWebhook(jwh *JiraWebhook, eventType string, format string, args ...inter
 	}
 }
 
-func (p *Plugin) GetWebhookURL(jiraURL string, teamID, channelID string) (subURL, legacyURL string, err error) {
-	cf := p.getConfig()
+func (wh *webhook) getConnection(p *Plugin, instance Instance, notification webhookUserNotification) (con *Connection, err error) {
+	var mattermostUserID types.ID
 
-	instanceID, err := p.ResolveWebhookInstanceURL(jiraURL)
+	// prefer accountId to username when looking up UserIds
+
+	if notification.jiraAccountID != "" {
+		mattermostUserID, err = p.userStore.LoadMattermostUserID(instance.GetID(), notification.jiraAccountID)
+	} else {
+		mattermostUserID, err = p.userStore.LoadMattermostUserID(instance.GetID(), notification.jiraUsername)
+	}
 	if err != nil {
-		return "", "", err
+		return
 	}
 
-	team, appErr := p.API.GetTeam(teamID)
-	if appErr != nil {
-		return "", "", appErr
+	// Check if the user has permissions.
+	con, err = p.userStore.LoadConnection(instance.GetID(), mattermostUserID)
+	if err != nil {
+		// Not connected to Jira, so can't check permissions
+		return
 	}
 
-	channel, appErr := p.API.GetChannel(channelID)
-	if appErr != nil {
-		return "", "", appErr
+	return
+}
+
+func (wh *webhook) CheckIssueWatchers(p *Plugin, instanceID types.ID) {
+	if len(wh.notifications) == 0 {
+		return
+	}
+
+	instance, err := p.instanceStore.LoadInstance(instanceID)
+	if err != nil {
+		// This isn't an internal server error. There's just no instance installed.
+		return
 	}
 
-	v := url.Values{}
-	secret, _ := url.QueryUnescape(cf.Secret)
-	v.Add("secret", secret)
-	subURL = p.GetPluginURL() + instancePath(routeAPISubscribeWebhook, instanceID) + "?" + v.Encode()
+	c, err := wh.getConnection(p, instance, webhookUserNotification{
+		jiraAccountID: wh.JiraWebhook.User.AccountID,
+		jiraUsername:  wh.JiraWebhook.User.Name,
+	})
+	if err != nil {
+		return
+	}
+	client, err2 := instance.GetClient(c)
```

Comment:
Why create `err2` here?

```suggestion
	client, err := instance.GetClient(c)
```